src/test/java/org/springframework/data/elasticsearch/core/RequestFactoryTests.java

```diff
@@ -562,6 +559,64 @@ void shouldSetStoredFieldsOnSearchRequest() {
 				.isEqualTo(Arrays.asList("last-name", "current-location"));
 	}
 
+	@Test // #1529
+	void shouldCreatePostReindexRequest() throws IOException, JSONException {
+		final String expected = "{\n" +
+				"    \"source\":{\n" +
+				"        \"remote\":{\n" +
+				"                \"username\":\"admin\",\n" +
+				"                \"password\":\"admin\",\n" +
+				"                \"host\":\"http://localhost:9200/elasticsearch\",\n" +
+				"                \"socket_timeout\":\"30s\",\n" +
+				"                \"connect_timeout\":\"30s\"\n" +
+				"            },\n" +
+				"        \"index\":[\"source_1\",\"source_2\"],\n" +
+				"        \"size\":5,\n" +
+				"        \"query\":{\"match_all\":{\"boost\":1.0}},\n" +
+				"        \"_source\":{\"includes\":[\"name\"],\"excludes\":[]},\n" +
+				"        \"slice\":{\"id\":1,\"max\":20}\n" +
+				"    },\n" +
+				"    \"dest\":{\n" +
+				"        \"index\":\"destination\",\n" +
+				"        \"routing\":\"routing\",\n" +
+				"        \"op_type\":\"create\",\n" +
+				"        \"pipeline\":\"pipeline\",\n" +
+				"        \"version_type\":\"external\"\n" +
+				"    },\n" +
+				"    \"max_docs\":10,\n" +
+				"    \"script\":{\"source\":\"Math.max(1,2)\",\"lang\":\"java\"},\n" +
+				"    \"conflicts\":\"proceed\"\n" +
+				"}";
+
+		Remote remote = Remote.builder("http", "localhost",9200)
+				.withPathPrefix("elasticsearch")
+				.withUsername("admin")
+				.withPassword("admin")
```

Comment:
nitpick: don't use the same value for different parameters. If in your code you erroneously set the usename value to both the userr and the password field the test would not fail.